File: src/main/java/destiny/astrology/SolarEclipseAnnularCentered.java
/**
 * Created by smallufo on 2017-11-08.
 */
package destiny.astrology;

public class SolarEclipseAnnularCentered extends SolarEclipseAnnular implements IEclipseCenter {

  protected final double centerBegin;

  protected final double centerEnd;

  public SolarEclipseAnnularCentered(double max, double partialBegin, double partialEnd, double totalBegin, double totalEnd, double centerBegin, double centerEnd, double annularBegin, double annularEnd) {
    super(max, partialBegin, partialEnd, totalBegin, totalEnd, annularBegin, annularEnd);

    this.centerBegin = centerBegin;
    this.centerEnd = centerEnd;
  }

  @Override
  public double getCenterBegin() {
    return centerBegin;
  }

  @Override
  public double getCenterEnd() {
    return centerEnd;
  }
}

File: src/main/java/destiny/astrology/eclipse/SolarEclipseAnnularCentered.kt
/**
 * Created by smallufo on 2017-11-08.
 */
package destiny.astrology.eclipse

open class SolarEclipseAnnularCentered(max: Double,
                                       partialBegin: Double,
                                       partialEnd: Double,
                                       totalBegin: Double,
                                       totalEnd: Double,
                                       override val centerBegin: Double,
                                       override val centerEnd: Double,
                                       annularBegin: Double,
                                       annularEnd: Double) :
  SolarEclipseAnnular(max, partialBegin, partialEnd, totalBegin, totalEnd, annularBegin, annularEnd), IEclipseCenter
